tests/s3_tester.c

```diff
@@ -231,6 +233,25 @@ static void s_s3_test_meta_request_telemetry(
     aws_s3_tester_unlock_synced_data(tester);
 }
 
+static void s_s3_test_meta_request_progress(
+    struct aws_s3_meta_request *meta_request,
+    const struct aws_s3_meta_request_progress *progress,
+    void *user_data) {
+
+    (void)meta_request;
+    AWS_ASSERT(meta_request);
+    AWS_ASSERT(progress);
+    AWS_ASSERT(user_data);
+
+    struct aws_s3_meta_request_test_results *meta_request_test_results = user_data;
+
+    aws_atomic_fetch_add(&meta_request_test_results->total_bytes_uploaded, (size_t)progress->bytes_transferred);
```

Comment:
worth to leave a comment about this `total_bytes_uploaded` only works for MPU, so that we won't surprise us again in the future.